vllm/entrypoints/openai/protocol.py

```diff
@@ -73,6 +73,8 @@ class ChatCompletionRequest(BaseModel):
     stop_token_ids: Optional[List[int]] = Field(default_factory=list)
     skip_special_tokens: Optional[bool] = True
     spaces_between_special_tokens: Optional[bool] = True
+    add_generation_prompt: Optional[bool] = True
```

Comment:
Can you explain this parameter? It is not part of the official API request protocol.

Reply:
Please see here: https://huggingface.co/docs/transformers/chat_templating#what-are-generation-prompts

It's part of the HF chat template definition. We had a discussion about it already up here: https://github.com/vllm-project/vllm/pull/1493#issuecomment-1806523808